drivers/spi/spi_sam0.c

```diff
@@ -77,6 +77,8 @@ static int spi_sam0_configure(const struct device *dev,
 	SercomSpi *regs = cfg->regs;
 	SERCOM_SPI_CTRLA_Type ctrla = {.reg = 0};
 	SERCOM_SPI_CTRLB_Type ctrlb = {.reg = 0};
+	SERCOM_SPI_CTRLC_Type ctrlc = {.reg = 0};
+	SERCOM_SPI_LENGTH_Type length = {.reg = 0};
```

Comment:
This is a little trick. You need to provide a patch that can be used by the devices that do not have the CTRLC reg.
For instance, this driver should still work with SAMD20.